jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jps.java

```diff
@@ -68,38 +83,52 @@ public static void main(String[] args) {
 		System.exit(rc);
 	}
 	
-	@SuppressWarnings("nls")
 	private static void parseArguments(String[] args) {
 		printApplicationArguments = false;
 		printJvmArguments = false;
 		noPackageName = true;
 		vmidOnly = false;
-		final String HELPTEXT = "jps: Print a list of Java processes and information about them%n"
-				+ "    -J: supply arguments to the Java VM running jps%n"
-				+ "    -l: print the application package name%n"
-				+ "    -q: print only the virtual machine identifiers%n"
-				+ "    -m: print the application arguments%n"
-				+ "    -v: print the Java VM arguments, including those produced automatically%n";
-		for (String a : args) {
-			switch (a) {
-			case "-l":
+
+		for (String arg : args) {
+			if (!arg.startsWith("-") || arg.length() == 1) {
+				System.out.printf("malformed argument '%s'%n%n", arg);
+				interruptWithHelpMessage();
+			}
+			arg = arg.substring(1);
+			for (char singleArg : arg.toCharArray()) {
+				processArgument(singleArg);
+			}
+		}
+	}
+
+	private static void processArgument(char arg) {
+		switch (arg) {
+			case 'l':
 				noPackageName = false;
 				break;
-			case "-m":
+			case 'm':
 				printApplicationArguments = true;
 				break;
-			case "-q":
+			case 'q':
 				vmidOnly = true;
 				break;
-			case "-v":
+			case 'v':
 				printJvmArguments = true;
 				break;
-				/* implicitly handle -h and -help via the default case */
-			default: 
-				System.out.printf(HELPTEXT);
-				System.exit(1);
+			case 'h':
+				System.out.printf(DESCRIPTION);
+				interruptWithHelpMessage();
+				break;
+			default:
+				System.out.printf("unsupported argument '-%c'%n%n", arg);
```

Comment:
Do we need different error message here?

Reply:
Pls expand why the message needs to be different. Are you referring to comparison with the RI?

Reply:
I will reuse `illegal argument`

Reply:
I think this should be `illegal argument` as well to match `RI`.

Reply:
Improved